### PR TITLE
Document backup partner maturity validation

### PR DESCRIPTION
**Pull request by ShellyKa13**
**What this PR does / why we need it**:
This doc describes how a backup partner can assess its compatibility with Kubevirt, either automatically or manually.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This doc refers to some stuff that is currently in a PR: https://github.com/kubevirt/kubevirt-velero-plugin/pull/247
hold until that gets merged.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
Great effort! Will you create a list of objects, which hold the state of a given VM too? So backup vendors know which objects they need to backup?

I'm thinking especially of hidden things like:
- PVC for persisted TPM state
- ControllerRevisions of Instancetypes/Preferences

But also regular objects like PVCs for disks, attached ConfigMaps, etc..

You could also add a test case for VMs with TPM persistence enabled.

File: docs/backup-restore-integration.md [hunk 1]
```diff
@@ -285,3 +285,146 @@ PersistentVolumeClaims ***owned by DataVolumes*** must have the following ***ann
...
+
+Create backup for the namespace, wait for it to complete successfully.
+
+Delete namespace
+
+Create restore from the backup and wait for completion.
+
+Check DV reaches succeeded state right away.
+
+check ConfigMap and Secret exists
```
Comment (mhenriks) [inline docs/backup-restore-integration.md:358]: Capital and period

File: docs/backup-restore-integration.md [hunk 2]
```diff
@@ -285,3 +285,146 @@ PersistentVolumeClaims ***owned by DataVolumes*** must have the following ***ann
...
+In this section, we will describe the different scenarios a backup partner should test in order to assess its compatibility with Kubevirt.
+
+### Prerequisites
+Cluster with the backup software and storage existing and Kubevirt + CDI installed.
+
+To validate there are two options:
+### Automatically
+This requires some extra effort.
+The tests are in our kubevirt-velero-plugin [repo](https://github.com/kubevirt/kubevirt-velero-plugin/tree/main/tests) marked with a label `PartnerComp`.
+You will need to adjust some env variables and the **backup and restore functions** in those tests.
```
Comment (mhenriks) [inline docs/backup-restore-integration.md:299]: Interesting idea to leverage the velerteo tests into a test framework of sorts.  Maybe we can have test args/env vars that are paths to backup/restore scripts that the framework executes?

**Review by mhenriks — COMMENTED** [review]
Thanks @ShellyKa13!  Very good start here!

File: docs/backup-restore-integration.md (hunk 2 above)
Comment (ShellyKa13) [reply docs/backup-restore-integration.md:299]: yeah sure, I can work on that :)

**Comment by ShellyKa13** [comment]
@0xFelix Hi felix thanks for your comment. The doc that I added this info to is basically what you described - a graph of objects that a vendor needs to make sure it backs up. We are working on a way to make sure this graph will remain updated.

**Comment by ShellyKa13** [comment]
/hold
This doc refers to some stuff that is currently in a PR: https://github.com/kubevirt/kubevirt-velero-plugin/pull/247
hold until that gets merged.

**Pull request merged**